cabal-install/src/Distribution/Client/CmdListBin.hs

```diff
@@ -151,9 +152,11 @@ listbinAction flags@NixStyleFlags{..} args globalFlags = do
     elaboratedPackage
         :: DistDirLayout
         -> ElaboratedSharedConfig
+        -> UnqualComponentName
+            -- TODO we should also include the component kind in addition to the name
```

Comment:
Could you resolve the TODO by making `singleComponentOrElse` return `ComponentName` instead of `UnqualComponentName`?

Reply:
This TODO only makes sense if a cabal package can have two components (possibly of different king) with with same unqualified component name. Is that at all possible? Otherwise, I think, this comment can be deleted.

Reply:
Yeah it isn't possible and using ComponentName wouldn't really make the current code cleaner since the components have to be unwrapped anyway